packages/logs/src/logger.ts

```diff
@@ -61,8 +61,8 @@ export function startLogger(
 ) {
   let globalContext: Context = {}
 
-  internalMonitoring.setExternalContextProvider(
-    () => deepMerge({ session_id: session.getId() }, globalContext, getRUMInternalContext() as Context) as Context
+  internalMonitoring.setExternalContextProvider(() =>
+    combine({ session_id: session.getId() }, globalContext, getRUMInternalContext())
```

Comment:
Nice! What about keeping deepMerge private to only use combine?

Reply:
I kept the export for the unit tests. But maybe we could replace `deepMerge` by `combine` in unit tests as well?

Reply:
sound good to me